BP-199 hiding states

julianweise opened the pull request:
Implemented feature to hide all cards containing products within hiddn states (defined by donught chart).

julianweise committed:
Hide cards with corresponding state when hidden in chart

Hides all cards holding a product with a state which has been hidden in the donught chart.

julianweise committed:
Added tests

Wrote corresponding tests for new functions introduced by this user story.

friedow commented:
Ich glaube du hast die Mock Änderungen die ich gestern vorgenommen hab nicht rebased. Dadurch versucht er hier alles neu zu comitten.

friedow reviewed (dismissed):
Rebase onto current changes to avoid commiting the whole project all over again.

friedow commented on src/DashboardView/Diagram/PieChart/PieChart.js:
```diff
@@ -28,20 +28,36 @@ class PieChart extends Component {
     componentDidMount() {
         const charWrapper = this.refs.canvas;
         if(charWrapper) {
+            const that = this;
```
Just don't.

friedow commented on src/DashboardView/Diagram/PieChart/PieChart.js:
```diff
@@ -28,20 +28,36 @@ class PieChart extends Component {
+                            //hide clicked legendCategory
+                            const index = legendItem.index;
+                            const lChart = this.chart;
+                            let i, ilen, meta;
```
Why are these variables allocated here before they are used?

friedow reviewed (commented):
When the outlook-like filtering function is implemented, we should move this to a text filter.

julianweise committed:
Code refactoring variable declaration

Corresponding to comments of Pull request I refactored naming and location of variable initialization to improve understandability of the code.
